File: APSS/_P_탑.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> heights) {
	vector<int> answer(heights.size());
	answer[0] = 0;
	for (int i = 1; i < heights.size(); i++) {
		answer[i] = 0;
		for (int j = i; j >= 0; j--) {
			if (heights[j] > heights[i]) {
				answer[i] = j + 1;
				break;
			}
		}
	}
	return answer;
}

File: ALGO/Algo_2.0/_P_탑.js
function solution(heights) {
  var answer = [];
  for (let i = heights.length - 1; i >= 0; i--) {
    for (let j = i; j >= 0; j--) {
      if (heights[i] < heights[j]) {
        answer.unshift(j + 1);
        break;
      }
      if (j == 0) answer.unshift(0);
    }
  }
  return answer;
}
